WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift

```diff
@@ -589,12 +593,56 @@ private extension DashboardViewController {
         .store(in: &subscriptions)
     }
 
-    private func dismissModalJustInTimeMessage() {
+    private func observeLocalAnnouncements() {
+        viewModel.$localAnnouncementViewModel
+            .compactMap { $0 }
+            .sink { [weak self] viewModel in
+                Task { @MainActor [weak self] in
+                    guard let self else { return }
+                    await self.dismissPossibleModals()
+                    let modalController = ConstraintsUpdatingHostingController(
+                        rootView: LocalAnnouncementModal_UIKit(
+                            onDismiss: {
+                                self.dismiss(animated: true)
+                            },
+                            viewModel: viewModel))
+
+                    self.localAnnouncementModalHostingController = modalController
+                    modalController.view.backgroundColor = .clear
+                    modalController.modalPresentationStyle = .overFullScreen
+                    self.present(modalController, animated: true)
+                }
+            }
```

Comment:
Super nit: Should we use `asyncMap` here to avoid the nested closure?
```suggestion
            .asyncMap { [weak self] viewModel in
                await self?.dismissPossibleModals()
                return ConstraintsUpdatingHostingController(
                        rootView: LocalAnnouncementModal_UIKit(
                            onDismiss: {
                                self?.dismiss(animated: true)
                            },
                            viewModel: viewModel))
              }
              .receive(on: DispatchQueue.main)
              .sink { [weak self] modalController in
                    self?.localAnnouncementModalHostingController = modalController
                    modalController.view.backgroundColor = .clear
                    modalController.modalPresentationStyle = .overFullScreen
                    self?.present(modalController, animated: true)
                }
```

Reply:
Thanks for the suggestion! Updated in https://github.com/woocommerce/woocommerce-ios/pull/10118/commits/cb3feb9ab1ad598fcffa21979214553707af2543